sdk/go/sqlite/sqlite.go

```diff
@@ -0,0 +1,133 @@
+// Package sqlite provides access to database stores within Spin
+// components.
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"errors"
+	"io"
+)
+
+// Open returns a new connection to the database.
+func Open(name string) *sql.DB {
+	return sql.OpenDB(&connector{name: name})
+}
+
+// Conn represents a database connection.
+type Conn struct {
+	_ptr uint32
+}
+
+// Close the connection.
+func (db *Conn) Close() error {
+	db.close()
+	return nil
+}
+
+// Query executes a query that may return rows, such as a SELECT.
+func (c *Conn) Query(query string, args []driver.Value) (driver.Rows, error) {
+	params := make([]any, len(args))
+	for i := range args {
+		params[i] = args[i]
+	}
+	return c.execute(query, params)
+}
+
+// QueryContext executes a query that may return rows, such as a SELECT.
+func (c *Conn) QueryContext(_ context.Context, query string, args []driver.Value) (driver.Rows, error) {
+	return c.Query(query, args)
+}
+
+// Exec isn't implemented. Use Query method.
+func (c *Conn) Exec(_ context.Context, _ string, _ []driver.NamedValue) (driver.Result, error) {
+	return nil, errors.New("Exec method not implemented")
+}
```

Comment:
I there a reason we're not implementing this?

Reply:
Correct me if I'm wrong... the returned object should be `LastInsertId` and `RowsAffected` which I don't think is returned from our implementation. https://pkg.go.dev/database/sql/driver#Result

Reply:
Ah you're correct. Sorry for the noise!

Reply:
I missed the note in the doc that says `Not every database or database driver may support this.` So I'll add it in